### PR TITLE
ElementComponent properties port to get/set 

### PR DESCRIPTION
**Pull request by kpal81xd**
Port of `ElementComponent` from using `Object.defineProperty` to using `get/set`

- [x] TS linter
- [x] API reference

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/framework/asset/asset-localized.js [hunk 1]
```diff
@@ -12,10 +12,17 @@ class LocalizedAsset extends EventHandler {
...
+        /** @type {number} */
+        // @ts-ignore
         this._defaultAsset = null;
+        /** @type {number} */
+        // @ts-ignore
         this._localizedAsset = null;
     }
 
+    /**
+     * @param {Asset | number} value - The asset or id.
```
Comment (willeastcott) [inline src/framework/asset/asset-localized.js:24]: I think this is not public API so consider adding `@ignore`.

File: src/framework/asset/asset-localized.js [hunk 2]
```diff
@@ -39,6 +46,9 @@ class LocalizedAsset extends EventHandler {
         return this._defaultAsset;
     }
 
+    /**
+     * @param {Asset | number} value - The asset or id.
```
Comment (willeastcott) [inline src/framework/asset/asset-localized.js:50]: I think this is not public API so consider adding `@ignore`.

File: src/framework/components/element/component.js [hunk 3]
```diff
@@ -1044,26 +1002,753 @@ class ElementComponent extends Component {
...
+     *      -  `[color="#0000ff"]blue text[/color] `
+     * 2.  `outline ` - override the element's  `outlineColor ` and  `outlineThickness ` properties.
+     * Example:
+     *      -  `[outline color="#ffffff" thickness="0.5"]text[/outline] `
+     * 3.  `shadow ` - override the element's  `shadowColor ` and  `shadowOffset ` properties.
+     * Examples:
+     *      -  `[shadow color="#ffffff" offset="0.5"]text[/shadow] `
+     *      -  `[shadow color="#000000" offsetX="0.1" offsetY="0.2"]text[/shadow]
+     *
+     * ` Note that markup tags are only processed if the text element's  `enableMarkup ` property is
```
Comment (willeastcott) [inline src/framework/components/element/component.js:1383]:
```suggestion
     *      -  `[shadow color="#000000" offsetX="0.1" offsetY="0.2"]text[/shadow]`
     *
     * Note that markup tags are only processed if the text element's  `enableMarkup ` property is
```

File: src/framework/components/element/component.js [hunk 4]
```diff
@@ -1044,26 +1002,753 @@ class ElementComponent extends Component {
...
+     * set to true.
```
Comment (willeastcott) [inline src/framework/components/element/component.js:1384]: This did not translate over from the original `@property` formatting correctly. Compare against the original...

File: src/framework/asset/asset-localized.js [hunk 5]
```diff
@@ -12,10 +12,17 @@ class LocalizedAsset extends EventHandler {
         this._autoLoad = false;
         this._disableLocalization = false;
 
+        /** @type {number} */
+        // @ts-ignore
```
Comment (willeastcott) [inline src/framework/asset/asset-localized.js:16]: This PR does introduce quite a lot of `@ts-ignore`s into the codebase. What's the plan here?

Comment (kpal81xd) [reply src/framework/asset/asset-localized.js:16]: A lot of these classes will have to be changed to properly represent their types. I dont like having them either but its whole separate job it itself and I dont want to bundle it together in this PR.

File: src/framework/components/element/component.js (hunk 4 above)
Comment (kpal81xd) [reply src/framework/components/element/component.js]: Fixed

File: src/framework/components/element/component.js [hunk 6]
```diff
@@ -806,7 +755,11 @@ class ElementComponent extends Component {
         this._setWidth(wr - wl);
 
         // update position
-        p.x = (this._localAnchor.z - this._localAnchor.x) - value - (this._calculatedWidth * (1 - this._pivot.x));
+        p.x =
+            this._localAnchor.z -
+            this._localAnchor.x -
+            value -
+            this._calculatedWidth * (1 - this._pivot.x);
```
Comment (willeastcott) [inline src/framework/components/element/component.js:762]: I'm a bit unsure about all these Prettier edits. I mean, is it _really_ prettier? 🤔  That said, I don't feel super-strongly, so I'll just hit approve...

**Review by willeastcott — APPROVED** [review]

File: src/framework/components/element/component.js (hunk 6 above)
Comment (kpal81xd) [reply src/framework/components/element/component.js:762]: Yea a lot of that im reverting its just for the new code ive added a lot of this i need to go back over and revert

**Pull request merged**